src/contracts/yas_pool.cairo

```diff
@@ -84,25 +185,304 @@ mod YASPool {
+                } else if step_tick_next > TickMath::MAX_TICK() {
+                    step_tick_next = TickMath::MAX_TICK();
+                };
+
+                // get the price for the next tick
+                let step_sqrt_price_next_X96 = TickMath::get_sqrt_ratio_at_tick(step_tick_next);
+
+                // compute values to swap to the target tick, price limit, or point where input/output amount is exhausted
+                let (ret_sqrt_price_X96, step_amount_in, step_amount_out, mut step_fee_amount) =
+                    SwapMath::compute_swap_step(
+                    state.sqrt_price_X96,
+                    if (zero_for_one && step_sqrt_price_next_X96 < sqrt_price_limit_X96)
+                        || (!zero_for_one && step_sqrt_price_next_X96 > sqrt_price_limit_X96) {
+                        sqrt_price_limit_X96
+                    } else {
+                        step_sqrt_price_next_X96
+                    },
+                    state.liquidity,
+                    state.amount_specified_remaining,
+                    self.fee.read()
+                );
+
+                state.sqrt_price_X96 = ret_sqrt_price_X96;
+
+                if exact_input {
+                    state
+                        .amount_specified_remaining -=
+                            IntegerTrait::<i256>::new(step_amount_in + step_fee_amount, false);
+                    state.amount_calculated = state.amount_calculated
+                        - IntegerTrait::<i256>::new(step_amount_out, false);
+                } else {
+                    state
+                        .amount_specified_remaining +=
+                            IntegerTrait::<i256>::new(step_amount_out, false);
+                    state.amount_calculated = state.amount_calculated
+                        + IntegerTrait::<i256>::new(step_amount_in + step_fee_amount, false);
+                };
+
+                // if the protocol fee is on, calculate how much is owed, decrement feeAmount, and increment protocolFee
+                if cache.fee_protocol > 0 {
+                    let delta: u256 = step_fee_amount / cache.fee_protocol.into();
+                    step_fee_amount -= delta;
+                    state.protocol_fee += delta.try_into().unwrap();
+                };
+
+                // update global fee tracker
+                if state.liquidity > 0 {
+                    state
+                        .fee_growth_global_X128 +=
+                            FullMath::mul_div(step_fee_amount, Q128, step_fee_amount);
+                };
+
+                // shift tick if we reached the next price
+                if state.sqrt_price_X96 == step_sqrt_price_next_X96 {
+                    // if the tick is initialized, run the tick transition
+                    if step_initialized {
+                        // crosses an initialized tick
+                        let mut liquidity_net: i128 = Tick::TickImpl::cross(
```

Comment:
In Uniswap they have a function returning a new value of `cache.tickCumulative` by calling `observeSingle`. Considering we are not using this function but we do need `tick_cumulative` should we consider modifying this variable anyway?

You can check the line [here](https://github.com/Uniswap/v3-core/blob/main/contracts/UniswapV3Pool.sol#L699)

Reply:
I removed the `tick_cumulative` variable since it is only used by the oracle